docs/configuration.md

```diff
@@ -0,0 +1,115 @@
+# Configuring and Tuning Cluster Logging
+
+OKD cluster logging is configurable by modifying the CustomResource deployed
+in the *openshift-logging* namespace.  The [sample](../hack/cr.yaml) resource
+defines a complete cluster logging deployment which includes all the subcomponents
+of the logging stack to collect, store and visualize logs.  The `cluster-logging-operator`
+watches the CustomResources and adjusts the logging deployment accordingly.
+
+## Management State
+Cluster logging is `Managed` by the `cluster-logging-operator`.  It can be placed
+into an unmanaged state allowing an administrator to assume full control of individual
+component configurations and upgrades by modifiying the `managementState` to `Unmanaged`.
+**Note:** An `Unmanaged` deployment will not receive updates until the operator is placed back into
+`Managed` state.
+```
+  spec:
+    managementState: "Managed"
+```
+
+## Component Images
+The image for each subcomponent can be overriden by modifying the appropriate
+environment variable in the `cluster-logging-operator` Deployment.  The variables
+define the full pull spec for the image:
+
+* `CURATOR_IMAGE`
+* `ELASTICSEARCH_IMAGE`
+* `FLUENTD_IMAGE`
+* `KIBANA_IMAGE`
+* `OAUTH_PROXY_IMAGE`
+* `RSYSLOG_IMAGE`
+
+**Note:** These changes may require the `cluster-version-operator` to be scaled to `0` so
+changes to the `cluster-logging-operator deployment are not reverted.
+
+## Common Configurations
+### Memory and CPU
+Each component specification allows for adjustments to both the CPU and 
+memory limits.  This can be added to the CustomResource by defining a `resources`
+block:
+```
+  spec:
+    logStore:
+      type: "elasticsearch"
+      elasticsearch:
+        resources:
+          limits:
+            cpu:
+            memory:
+          requests:
+            cpu:
+            memory:
+```
+
+## Data Aggregation and Storage
+An Elasticsearch cluster is responsible for log aggregation.  Following is a sample 
+of the spec:
+```
+  spec:
+    logStore:
+      type: "elasticsearch"
+      elasticsearch:
+        replicas: 1
+        storage:
+          emptyDir: {}
+        dataReplication: "NoReplication"
+```
+### Backing storage
+|Storage Source | Description |
+|----- | ----- |
+|`emptyDir`|Ephemiral storage|
+|`hostPath`|Storage directly mounted from the cluster host node. Requires `hostaccess` scc granted to the ServiceAccount and appropriate directory permissions, SELINUX labels|
+|`volumeClaimTemplate`|The prefix to use to generate PVCs to support the Elasticsearch cluster|
+|`persistentVolumeClaim`|How do we use this with an N node cluster?|
+
+### Data Replication
+|Policy | Description |
+|----- | ----- |
+|FullReplication | A single copy of each shard. Logs are always available and recoverable as long as at least two nodes exist  |
+|PartialReplication| ?? |
```

Comment:
@richm @ewolinetz @josefkarasek Given how we actually apply replication configuration we should remove this option as it doesn't make any sense.  The current CLO CR does not make it possible to selectively replicate certain indices but not others.